picolib/build-arm.sh

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/env bash
+
+set -x
+
+PICOLIB_SRC_DIR=$1
+PICOLIB_INSTALL_DIR=$2
+
+# We want to end up with newlib compiled for at least the `v6-m/nofp`,
+# `v7-m/nofp`, and `v7e-m/nofp` architectures. For that to happen the
+# arm-none-eabi-gcc compiler must have been compiled with multilib support for
+# those architectures. We verify that here before actually building anything.
+if ! arm-none-eabi-gcc --print-multi-lib | grep -q 'v6-m/nofp'; then
+  echo "ERROR: arm-none-eabi-gcc not configured with multilib support for v6-m/nofp"
+  exit -1
+fi
+if ! arm-none-eabi-gcc --print-multi-lib | grep -q 'v7-m/nofp'; then
+  echo "ERROR: arm-none-eabi-gcc not configured with multilib support for v7-m/nofp"
+  exit -1
+fi
+if ! arm-none-eabi-gcc --print-multi-lib | grep -q 'v7e-m/nofp'; then
+  echo "ERROR: arm-none-eabi-gcc not configured with multilib support for v7e-m/nofp"
+  exit -1
+fi
+
+# Picolibc Configure Flags
+#
+# picolib=false             # Do not use picolib sbrk. We need to use the memop syscalls.
+# picocrt=false             # Do not build crt0, we have our own.
+# picocrt-lib=false         # Do not package crt0.
+# tinystdio=false           # Do not use AVR tinystdio, which uses a different underlying interface (aka stdout object).
+# newlib-stdio64=false      # Do not include 64 bit APIs.
+# newlib-reent-small=true   # ?
```

Comment:
```suggestion
# newlib-reent-small=true   # Disable reentrancy support to save space
```

This option matches what we do with newlib (namely, turning on `newlib-nano` flags). 

Here's a good explainer for more: https://www.codeinsideout.com/blog/freertos/reentrant/#project-setup

---

Reading that and thinking for not enough seconds, I wonder if we have some potential issues around reentrancy in libtock in the absence of `yield_for`; if multiple event-driven upcalls called printf for example, I think things as-written would break [in newlib and in picolib; this isn't a problem for this PR].

1. upcall_[tempReading] --> _in upcall handler_, printf("temp is ...") --> ..`<stdio>`.. --> putnstr userspace `yield_for` for upcall to finish
2. upcall_[lightReading] --> _in upcall handler_, printf("light is ..." --> ..`<stdio>`.. [BAD, prior printf not done; state trampled]

This goes away with kernel-supported `yield_for` though, so pushing on that fix feels of greater value than stressing about this. Still, it's a limitation we should note somewhere somehow probably [again; not this PR's problem]

Reply:
Yes I don't think there is any robust way to mix async and sync code currently.